src/test/java/org/sonarsource/plugins/example/rules/ForbiddenKeywordsLocatorTest.java

```diff
@@ -0,0 +1,88 @@
+package org.sonarsource.plugins.example.rules;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.sonarsource.plugins.example.fixtures.InputFileFixtures.aFileWithContent;
+
+import java.io.IOException;
+import java.util.List;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.sonar.api.batch.fs.TextRange;
+
+public class ForbiddenKeywordsLocatorTest {
+
+  @Before
+  public void setUp() {
+    locator = new ForbiddenKeywordsLocator("foo");
+  }
+
+  @Test
+  public void it_should_not_locate_forbidden_keyword_if_file_is_empty() throws IOException {
+    List<TextRange> occurrences = locator.locateAllIn(aFileWithContent(""));
+
+    assertThat(occurrences).isEmpty();
+  }
+
+  @Test
+  public void it_should_locate_forbidden_keyword_if_file_contains_only_the_keyword() throws IOException {
+    List<TextRange> occurrences = locator.locateAllIn(aFileWithContent("foo"));
+
+    assertThat(occurrences).hasSize(1);
```

Comment:
all this block can possibly be replaced by something like:
```
assertThat(occurrences)
  .extracting(r -> r.start().line(), r -> r.start().offset(), ...)
  .containsOnly(tuple(1, 0, 1, 3));
```
In case of failure of the assertion, the error message will give you all the informations